src/main/kotlin/three/consulting/epoc/service/CustomerService.kt

```diff
@@ -33,6 +34,14 @@ class CustomerService(private val customerRepository: CustomerRepository) {
         else throw UnableToUpdateCustomerException()
     }
 
+    fun deleteCustomer(customerId: Long) {
+        try {
+            customerRepository.deleteById(customerId)
+        }catch (e: EmptyResultDataAccessException) {
+            throw UnableToDeleteCustomerException(customerId)
```

Comment:
When catching an expcetion and throwing a new one, you can include the original in the new one and pass it along to the RunTimeException for improved logging. So:
```kotlin
throw UnableToDeleteCustomerException(customerId, e)
```